src/ethernet/eth.rs

```diff
@@ -648,31 +638,58 @@ pub unsafe fn new_unchecked(
     });
 
     // MAC layer
-    let mut mac = EthernetMAC { eth_mac };
-    mac.phy_reset();
-    mac.phy_init(); // Init PHY
+
+    // Set the MDC clock frequency in the range 1MHz - 2.5MHz
+    let hclk_mhz = clocks.hclk().0 / 1_000_000;
```

Comment:
I feel like we should be doing this by converting the hertz into megahertz here via an explicit mhz() call but it doesn't seem like that is implemented as of now so it probably doesn't matter. However doing all unit conversion through the type system instead of like this seems like a better idea to me since it's probably going to save us from doing some mistakes in the future.

Reply:
Hmm, I'd want to be able to write something like:

```rust
let hclk_mhz: MegaHertz = clocks.hclk().into();

match hclk_mhz.0 { ... }
```

But that's a "lossy" conversion that isn't currently implemented. I had a look at what [embedded-time does](https://github.com/FluenTech/embedded-time/blob/master/src/rate.rs#L585): beneath much indirection it implements conversions in both directions, and actually conversions like MegaHertz -> Hertz are fallible (`TryFrom`) because they could overflow, whilst the opposite Hertz -> MegaHertz is infallible.

In any case I don't see a better way of addressing it in this PR; but the discussion could be continued in another issue.